### ISSUE TITLE
Redirect HTTP traffic to HTTPS

### ISSUE THREAD

**Issue by assafharel** [PeerSay/Atlas#47]
Our cloudflare account expects an https traffic from peersay.herokuapp.com.
See this short thread regarding implementation:
http://stackoverflow.com/questions/26960661/expressjs-redirecting-to-https-using-cloudflare-flexible-ssl

Assaf

**Comment by Pavek** [comment]
>>Our cloudflare account expects an https traffic from peersay.herokuapp.com.
I don't think this is true. CloudFalre is client to our Heroku host; and client (not server) is responsible for establishing secure/unsecure connection. I think the truth is in these words (from stackoverflow article):
CloudFlare's server is actually talking to your server via unsecured http.

>>See this short thread regarding implementation
This Stackoverflow article actually solves a different problem: how to tell browser to switch to HTTPS from server behind Reverse-Proxy (CloudFlare in our case) -- for certain (or all) URLs of the website.

Having that our server sees only HTTP connections from CloudFlare, the solution is well-know use of x-forwarded-proto.

But in our case I don't see why is this required -- the 301-redirect to HTTPS already happens:
http://gyazo.com/377e330bc4972f95438263825e27e9e3
(I believe this is because of some settings of CludFlare).

Please elaborate what exactly is the use case we need to fix.

**Comment by assafharel** [comment]
Hi Pavel,

These are 2 separated things.
1. I already set Cloudflare to automatically switch (using 301) from HTTP to HTTPS. They also provided the certificate for free (their cheap solution is to add us a domain to one of their multi-domain certificates)

2. In our cloudflare account I chose a HTTPS connection between cloudflare and heroku. I sepcifically configured cloudflare to connect to https://peersay.herokuapp.com.

3. Now, from you I ask to force SSL as explained in the stackoverflow issue:
//FORCE SSL
app.use(function(req, res, next) {
  if(req.headers['x-forwarded-proto']==='http') {
    return res.redirect(['https://', req.get('Host'), req.url].join(''));
  }
  next();
});
so that anything will be forced to SSL (or explain why it shouldn't be so).

Thanks,
Assaf
